cli/dcoscli/marathon/main.py

```diff
@@ -596,15 +599,58 @@ def _update(app_id, properties, force):
     client = marathon.create_client()
 
     # Ensure that the application exists
-    client.get_app(app_id)
+    current_app = client.get_app(app_id)
+
+    schema = _app_schema()
+    app_resource = _parse_properties(properties, schema)
+    _validate_update(current_app, app_resource, schema)
 
-    app_resource = _parse_properties(properties, _app_schema())
     deployment = client.update_app(app_id, app_resource, force)
 
     emitter.publish('Created deployment {}'.format(deployment))
     return 0
 
 
+def _validate_update(current_resource, properties, schema):
+    """
+    Validate resource ("app" or "group") update
+
+    :param current_resource: Marathon app definition
+    :type current_resource: dict
+    :param properties: resource JSON
+    :type properties: dict
+    :param schema: JSON schema used to verify properties
+    :type schema: dict
+    :rtype: None
+    """
+    updated_resource = _clean_up_properties(current_resource.copy())
+    updated_resource.update(properties)
+
+    errs = util.validate_json(updated_resource, schema)
+    if errs:
+        raise DCOSException(util.list_to_err(errs))
+
+
+def _clean_up_properties(properties):
```

Comment:
Let's change this method name to something more descriptive. Maybe `_clean_up_app_definition`? Is app definition the correct term? The schema has apps and groups, no? Does Marathon have a name for that schema?